evaluation/tasks/auto_task.py

```diff
@@ -1,25 +1,76 @@
 from abc import ABC, abstractmethod
+from typing import Dict
 import os
 
-from evaluation.utils.io import save_json
+from transformers import AutoTokenizer
+
+from evaluation.utils.io import save_json, load_json
+from evaluation.models.loader import load_model
 
 
 class AutoTask(ABC):
-    def __init__(self, tokenizer, model, device):
-        self.tokenizer = tokenizer
+    def __init__(
+        self, 
+        model,
+        tokenizer,
+        device, 
+        english_only: bool, 
+    ):
         self.model = model
+        self.tokenizer = tokenizer
         self.device = device
         self.metrics = {}
-
+        self.task_config = self.load_task_args(english_only)
+    
     @classmethod
-    def from_task_name(cls, task_name: str, tokenizer, model, device):
+    def _get_task(cls, task_name):
         all_tasks = cls.__subclasses__()
         for task in all_tasks:
             if task.get_display_name() == task_name:
-                return task(tokenizer=tokenizer, model=model, device=device)
-        
+                return task
         raise ValueError(f'Invalid task: {task_name}')
 
+    @classmethod
+    def from_task_name(
+        cls, 
+        task_name: str, 
+        model, 
+        tokenizer,
+        device, 
```

Comment:
For consistency, let's provide type hinting for all the params -- ie `model: PreTrainedModel`, `tokenizer: PreTrainedTokenizerFast` and `device: torch.device`.

Reply:
Yes, type hinting is a good idea. Thanks for the heads up!